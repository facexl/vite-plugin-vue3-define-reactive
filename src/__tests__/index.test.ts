import { transformDefineReactiveMacro } from '../index'

describe('transformDefineReactiveMacro', () => {
    test('should expose all params key top level declarations', () => {
      const content = transformDefineReactiveMacro(`
        <script setup>
            const state = defineReactive({
                showApp:false,
                loading:false
            })
        </script>
      `,{
            needImport:true
        })
      expect(content).toMatch('const {showApp,loading} = toRefs(state)')
    })
    test('should import missing dependencies', () => {
        const content = transformDefineReactiveMacro(`
          <script setup>
              import { toRefs } from 'vue'
              defineReactive({
                  showApp:false,
                  loading:false
              })
          </script>
        `,{
              needImport:true
          })
        expect(content).toMatch(`import { reactive } from 'vue'`)
    })
    test('should auto declare new variable', () => {
        const content = transformDefineReactiveMacro(`
          <script setup>
              defineReactive({
                  showApp:false,
                  loading:false
              })
          </script>
        `,{})
        expect(content).toMatch(`auto_identifier__v_15`)
    })
    test('ts', () => {
        const content = transformDefineReactiveMacro(`
        <script setup lang="ts">
        const state = defineReactive({
            a:2 as number
        } as any) as any
        let x,y,z;
        const r = function(){}
        const r2 = (function(){})()
        let a  = 1
        const b = 2
        function c() {}
        class d {}
        defineReactive({
            r:1
        }) as fuck
    </script>
        `,{
        })
        expect(content).toMatch(`const {a} = toRefs(state)`)
    })
    test('example',()=>{
        const content = transformDefineReactiveMacro(`<template>
        <div>{{a}}</div>
    </template>
    <script setup>
    defineReactive({
        a:1,
        b:2
    })
    </script> `,{
            needImport:true
        })
        console.log(content)
    })
})