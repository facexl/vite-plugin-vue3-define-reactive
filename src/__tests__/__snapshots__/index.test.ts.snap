// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformDefineReactiveMacro should auto declare new variable 1`] = `
"<script setup>
              
 const auto_identifier__v_15=reactive({
                  showApp:false,
                  loading:false
              })
          
 const {showApp,loading} = toRefs(auto_identifier__v_15)
</script>"
`;

exports[`transformDefineReactiveMacro should expose all params key top level declarations 1`] = `
"<script setup>
 import { toRefs,reactive } from 'vue' 
 
            const state = reactive({
                showApp:false,
                loading:false
            })
        
 const {showApp,loading} = toRefs(state)
</script>"
`;

exports[`transformDefineReactiveMacro should import missing dependencies 1`] = `
"<script setup>
 import { reactive } from 'vue' 
 
              import { toRefs } from 'vue'
              
 const auto_identifier__v_58=reactive({
                  showApp:false,
                  loading:false
              })
          
 const {showApp,loading} = toRefs(auto_identifier__v_58)
</script>"
`;

exports[`transformDefineReactiveMacro ts 1`] = `
"<script setup lang=\\"ts\\">
        const state = reactive({
            a:2 as number
        } as any) as any
        let x,y,z;
        const r = function(){}
        const r2 = (function(){})()
        let a  = 1
        const b = 2
        function c() {}
        class d {}
        
 const auto_identifier__v_267=reactive({
            r:1
        }) as any
    
 const {r} = toRefs(auto_identifier__v_267)

 const {a} = toRefs(state)
</script>"
`;
